### ISSUE TITLE
Reinstate record from line in from Airtime 2.5.x

### ISSUE THREAD

**Issue by comiconomenclaturist** [libretime/libretime#42]
A lot of the code seems to still exist, including recorder.py and some of the PHP functions.

**Comment by hairmare** [comment]
I think [2.5.x:airtime_mvc/application/views/scripts/schedule/add-show-form.phtml#L33-L37](https://github.com/LibreTime/libretime/blob/archive/airtime/2.5.x/airtime_mvc/application/views/scripts/schedule/add-show-form.phtml#L33-L37) is all that's missing.

**Comment by comiconomenclaturist** [comment]
Yup I found that bit of code for entering shows to be recorded, but I don't think the ShowRecorder function in recorder.py is being called. At least no ecasound process starts!

**Comment by hairmare** [comment]
Ok, i removed "Fixes" annotation from the comment on #43. If this should close when we merge that, please reopen it if we haven't found the issue.

Do you know at what version the recorder last worked? It seems to still get started [in pypo](https://github.com/LibreTime/libretime/blob/276a69aa1bef527260efc170d7caab7d04b4ca13/python_apps/pypo/pypo/__main__.py#L298-L300).

**Comment by hairmare** [comment]
I'm at the point where it calls ecasound on my vagrant box and then fails miserably. Most likely it is some issue with file permissions or a missing folder somewhere. Also, after adding a new recording i t takes up to an hour for the recorder to pick up on it so keep that in mind if you want to test this.

**Comment by hairmare** [comment]
A quick update on this... After biting my teeth out on how to get the recording from ecasound uploaded to the API for inclusion in the library through analyzer (without loosing all the metadata concerning that it's a pre-recording). 

We should probably rewrite the code so that the file entry in the db gets pre-populated so we can pass its id to pypo-recorder when requesting the recording. pypo-recorder would then be able to just PUT an update to the file resource, that already contains the needed metadata. pypo-recorder would need to add proper id3 tags as well (I need to re-check if it does anything smart yet).

On the MVC side of things the models need to be adapted to store these file resources in a fashion similar to how podcast episodes and watched files are handled. It also makes sense to clean up the ui integration points a bit and keep the recordings nicely split from the main library in a fashion similar to other things like podcasts and streams.

Sorry about the braindump but I just realized that I need somewhere to put this. I'm currently focussing on a largish js refactor but I hope I'll find some time to look into this.

**Comment by GroundTruthNZ** [comment]
I managed to get the recording part working consistently from a stream using a slightly modified version of @JohnnyC1951's [script](https://github.com/LibreTime/libretime/issues/291#issuecomment-379830893) 

The audio files get loaded into the library and appear to be good.  Although they aren't attached to the show or future scheduled shows it will still be a very useful feature (I.e to record a scheduled show, but without the rebroadcast function).

Attempting to record from line-in is always going to be a challenge (or impossible) with cloud-based, or docker based instances.  I suggest re-enabling the record function from Stream 1 (or the master or show source instead).   The rebroadcasting functionality could be turned on later once we've figured out how to handle it...

POC code is here:
https://github.com/GroundTruthNZ/libretime/commit/a15ef48f3dcb87474068bdb6c1e2f327b85b4019

The source in line 93 in the recorder.py script is hard coded here but could presumably be dynamic .
